Src/java/engine-fhir/src/main/java/org/opencds/cqf/cql/engine/fhir/converter/Dstu2FhirTypeConverter.java

```diff
@@ -227,7 +228,35 @@ public IBase toFhirTuple(Tuple value) {
             return null;
         }
 
-        throw new NotImplementedException("can't convert Tuples");
+        var parameters = new Parameters();
+        if (value.getElements().isEmpty()) {
+            return parameters;
+        }
+
+        var param = parameters.addParameter();
+        for (String key : value.getElements().keySet()) {
+            var part = param.addPart();
+            part.setName(key);
+            var element = value.getElements().get(key);
+            if (element == null) {
+                part.addExtension()
+                        .setUrl(DATA_ABSENT_REASON_EXT_URL)
+                        .setValue(new CodeType(DATA_ABSENT_REASON_UNKNOWN_CODE));
+            } else if (element instanceof Collection && ((Collection<?>) element).isEmpty()) {
+                part.addExtension().setUrl(EMPTY_LIST_EXT_URL).setValue(new BooleanType(true));
+            } else {
+                var result = toFhirType(element);
```

Comment:
What happens in the case that the element is a Tuple or List? The expected behavior is that these would be subparts, with Lists represented as repeating elements: https://hl7.org/fhir/uv/cql/Parameters-cql-typemappingexampleresult.html

Reply:
If I'm reading the code right, what it will actually do is put this in as a Parameters resource

Reply:
This isn't mapping all the evaluation results for a Library; it's mapping arbitrary (potentially recursive) values. There's no way to know that you need to lift or flatten the way "Using CQL with FHIR" IG specifies without that larger context. You'd need a second step to flatten the results. We don't even have the expression name here.